mmv1/third_party/terraform/services/bigtable/resource_bigtable_instance.go

```diff
@@ -258,25 +258,26 @@ func resourceBigtableInstanceRead(d *schema.ResourceData, meta interface{}) erro
 
 	ctxWithTimeout, cancel := context.WithTimeout(ctx, d.Timeout(schema.TimeoutRead))
 	defer cancel()
-	instance, err := c.InstanceInfo(ctxWithTimeout, instanceName)
-	if err != nil {
-		if tpgresource.IsNotFoundGrpcError(err) {
-			log.Printf("[WARN] Removing %s because it's gone", instanceName)
-			d.SetId("")
-			return nil
-		}
+	instancesResponse, err := c.Instances(ctxWithTimeout)
+	instance, stop, err := getInstanceFromResponse(instancesResponse, instanceName, err, d)
+	if stop {
 		return err
 	}
 
 	if err := d.Set("project", project); err != nil {
 		return fmt.Errorf("Error setting project: %s", err)
 	}
 
-	clusters, err := c.Clusters(ctxWithTimeout, instance.Name)
+	clusters, err := c.Clusters(ctxWithTimeout, instanceName)
```

Comment:
okay, so it looks like we're assuming that if the instance is impacted, the clusters call will always show that the instance is unavailable - that seems like a reasonable assumption... this would cause a very strange panic if a cluster recovered in between the two calls, though. even though it's unlikely - what would you think about returning an "unavailable zones" error directly from getInstanceFromResponse instead of falling through to the clusters call?

Reply:
My issue is with getting the instance zones.
It's possible I don't get the instance and I get "unavailable zones" > 0. In that case, I don't know if the instance doesn't exist OR the instance exists, in unavailable zones. Only the Clusters() call knows that information, so I delay this decision.

Reply:
that makes sense, thanks!